src/transformers/pipelines/text_to_audio.py

```diff
@@ -105,13 +128,22 @@ def preprocess(self, text, **kwargs):
         return output
 
     def _forward(self, model_inputs, **kwargs):
+        forward_params = kwargs["forward_params"]
+        generate_kwargs = kwargs["generate_kwargs"]
+
         # we expect some kwargs to be additional tensors which need to be on the right device
-        kwargs = self._ensure_tensor_on_device(kwargs, device=self.device)
+        forward_params = self._ensure_tensor_on_device(forward_params, device=self.device)
 
         if self.model.can_generate():
-            output = self.model.generate(**model_inputs, **kwargs)
+            # we expect some kwargs to be additional tensors which need to be on the right device
+            generate_kwargs = self._ensure_tensor_on_device(generate_kwargs, device=self.device)
+
+            # generate_kwargs get priority over forward_params
+            forward_params.update(generate_kwargs)
+
+            output = self.model.generate(**model_inputs, **forward_params)
         else:
-            output = self.model(**model_inputs, **kwargs)[0]
+            output = self.model(**model_inputs, **forward_params)[0]
```

Comment:
Should we error out if `generate_kwargs` are passed but the model doesn't generate? We could then nudge the user to use `forward_kwargs` if using a forward-only model in the error message. WDYT?